web/src/main/java/com/marklogic/hub/web/web/LoadDataController.java

```diff
@@ -0,0 +1,68 @@
+package com.marklogic.hub.web.web;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.web.service.ArtifactService;
+import com.marklogic.hub.web.service.MlRestErrorHandlerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+@RequestMapping("/api/current-project/loadData")
+@EnableAsync
+public class LoadDataController {
+    @Autowired
+    private ArtifactService artifactService;
+
+    @Autowired
+    private MlRestErrorHandlerService mlRestErrorHandlerService;
+
+    private String artifactType = "loadData";
+
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity<JsonNode> getLoadDataConfigurations() {
+        return mlRestErrorHandlerService.handleRestException(() ->
```

Comment:
I think this is better handled via Spring MVC's HandlerExceptionResolver mechanism - here's one article on it - https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc#handlerexceptionresolver . You can register a global error handler for a FailedRequestException. Then controllers don't need to worry about this, they can just invoke the appropriate data service interface.

Reply:
Registered FailedRequestException with our custom exception handler instead.